receiver/awscontainerinsightreceiver/internal/neuron/neuron_empty_metric_decorator.go

```diff
@@ -0,0 +1,159 @@
+}
+
+type EmptyMetricDecorator struct {
+	NextConsumer consumer.Metrics
+	Logger       *zap.Logger
+}
+
+func (ed *EmptyMetricDecorator) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: true,
+	}
+}
+
+func (ed *EmptyMetricDecorator) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rs := rms.At(i)
+		ilms := rs.ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ils := ilms.At(j)
+			metrics := ils.Metrics()
+
+			neuronHardwareInfo, neuronHardwareInfoFound := findNeuronHardwareInfo(metrics)
+			if neuronHardwareInfoFound {
+				ed.addEmptyMetrics(neuronHardwareInfo, metrics)
+			}
+		}
+	}
+	return ed.NextConsumer.ConsumeMetrics(ctx, md)
+}
+
+func (ed *EmptyMetricDecorator) addEmptyMetrics(hardwareInfo pmetric.Metric, metrics pmetric.MetricSlice) {
+	var metricFoundMap = make(map[string]bool)
+	for k := range attributeConfig {
+		metricFoundMap[k] = false
+	}
+
+	for k := 0; k < metrics.Len(); k++ {
+		m := metrics.At(k)
+		if _, ok := metricFoundMap[m.Name()]; ok {
+			metricFoundMap[m.Name()] = true
+		}
+	}
+
+	for k, v := range metricFoundMap {
+		if !v {
+			if strings.Contains(k, "core") {
+				populateCoreMetrics(metrics, k, attributeConfig[k], hardwareInfo)
+			} else {
+				populateNonCoreMetrics(metrics, k, attributeConfig[k], hardwareInfo)
+			}
+		}
+
+	}
+}
+
+// method populates per non-core metrics, thus empty metrics are added only once per node
+func populateNonCoreMetrics(metrics pmetric.MetricSlice, metricName string, attributesToAdd []string, hardwareInfo pmetric.Metric) {
+	metricToAdd := createNewMetricFromHardwareInfo(hardwareInfo, metricName)
+	metricBody := metricToAdd.Gauge().DataPoints().At(0)
+
+	for _, attribute := range attributesToAdd {
+		metricBody.Attributes().PutStr(attribute, nonCoreAttributeValues[attribute])
+	}
+
+	metricToAdd.CopyTo(metrics.AppendEmpty())
+}
+
+// method populates per core metrics, thus empty metrics are added per core
+func populateCoreMetrics(metrics pmetric.MetricSlice, metricName string, attributesToAdd []string, hardwareInfo pmetric.Metric) {
+	neuronCoresPerDevice, foundCoresPerDevice := getNeuronCoresPerDevice(hardwareInfo)
+	neuronDeviceCount, foundDeviceCount := getNeuronDeviceCount(hardwareInfo)
+	if foundCoresPerDevice && foundDeviceCount {
+		for coreIndex := 0; coreIndex < neuronCoresPerDevice; coreIndex++ {
+			for deviceNum := 0; deviceNum < neuronDeviceCount; deviceNum++ {
+				metricToAdd := createNewMetricFromHardwareInfo(hardwareInfo, metricName)
+				metricBody := metricToAdd.Gauge().DataPoints().At(0)
+
+				for _, attribute := range attributesToAdd {
+					if attribute == neuronCoreAttributeKey {
+						metricBody.Attributes().PutStr(attribute, strconv.Itoa(coreIndex*neuronDeviceCount+deviceNum))
+					}
+					if attribute == neuronDeviceAttributeKey {
+						metricBody.Attributes().PutStr(attribute, strconv.Itoa(deviceNum))
+					}
+				}
```

Comment:
Minor: having it in a for loop is adding no value, you might as well just have two statements

```
metricBody.Attributes().PutStr(neuronCoreAttributeKey, strconv.Itoa(coreIndex*neuronDeviceCount+deviceNum))
metricBody.Attributes().PutStr(neuronDeviceAttributeKey, strconv.Itoa(deviceNum))
```